src/Core/Domain/Product/Command/AbstractProductCommand.php

```diff
@@ -0,0 +1,568 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\ManufacturerConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\ValueObject\ManufacturerId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\Condition;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\Price\CostPrice;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\MetaData\FriendlyUrl;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\MetaData\MetaDescription;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\MetaData\MetaKeywords;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\MetaData\MetaTitle;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductName;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\RedirectionPage\RedirectionPageInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\Price\RetailPrice;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\RedirectionPage\TypedRedirectionPageInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\Price\UnitPrice;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\Reference\Ean13;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\Reference\Isbn;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\Reference\Reference;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\Reference\Upc;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\Visibility;
+
+/**
+ * Holds the abstraction of common product data which does not depend from product type.
+ */
+abstract class AbstractProductCommand
+{
+    /**
+     * @todo: I need defaultLanguage validation in handler
+     * @var ProductName[]
```

Comment:
the only useful purpose I see that it servers as length validator - do you think we don't need to do such validations?

Reply:
I also think it's overkill. If we had to check for special characters and whatnot then maybe it could be justified, but fundamentally it's just a string.